Sources/NIO/BlockingIOThreadPool.swift

```diff
@@ -153,6 +155,31 @@ public final class BlockingIOThreadPool {
     }
 }
 
+public extension BlockingIOThreadPool {
+    /// Runs the submitted closure if the thread pool is still active, otherwise fails the promise.
+    /// The closure will be run on the thread pool so can do blocking work.
+    ///
+    /// - parameters:
+    ///     - eventLoop: The `EventLoop` to create the returned `EventLoopFuture` on.
+    ///     - body: The closure which performs some blocking work to be done on the thread pool.
+    /// - returns: An `EventLoopFuture` fulfilled with the result (or error) of the passed closure.
+    public func runIfActive<T>(eventLoop: EventLoop, _ body: @escaping () throws -> T) -> EventLoopFuture<T> {
+        let promise: EventLoopPromise<T> = eventLoop.newPromise()
+        self.submit { shouldRun in
+            guard case shouldRun = BlockingIOThreadPool.WorkItemState.active else {
+                promise.fail(error: ChannelError.ioOnClosedChannel)
```

Comment:
@normanmaurer yes, that is indeed very odd. However that is what the other methods do so I copied it. Happy to fix but then we're slightly inconsistent...

Reply:
hmm ok... We should have spotted this before :( I guess we need to fix this in a next major

Reply:
@normanmaurer well, we _could_ just be different in this particular method. That wouldn't be breaking API as it's a new API. It would just behave slightly differently than the `read` apis.